### ISSUE TITLE
Discord Fails To Display Anything With Nushell

### ISSUE THREAD

**Issue by ColdIce1605** [nushell/nushell#5221]
### Describe the bug

Discord just has the background being displayed.

If you need anymore additional information I'll be happy to provide it. 

### How to reproduce

1.
Set nu as default shell using 
`chsh -s /bin/nu`
2.
Log out and log back in
3.
Launch discord
4.
Has blank screen

### Expected behavior

I expect nu to respect 
`#!/usr/bin/env bash`
but it doesn't seem to as if it just handed it to bash I would expect it to work idk if this isn't apart of nu syntax but it would be awesome to both use discord with nu as my default shell. 

### Screenshots

With Bash:
https://i.imgur.com/hzetF5Z.png
With Nushell:
https://i.imgur.com/N3ttcMi.png

### Configuration

| key                | value                                     |
| ------------------ | ----------------------------------------- |
| version            | 0.61.0                                    |
| branch             |                                           |
| tag                |                                           |
| short_commit       | 3783c19d                                  |
| commit_hash        | 3783c19d02f4a3b2f35cd26b79daabcb70b04880  |
| commit_date        | 2022-04-12 20:38:15 +00:00                |
| build_os           | linux-x86_64                              |
| rust_version       | rustc 1.60.0 (Arch Linux rust 1:1.60.0-1) |
| cargo_version      | cargo 1.60.0                              |
| pkg_version        | 0.61.0                                    |
| build_time         | 2022-04-12 23:20:08 +00:00                |
| build_rust_channel | release                                   |
| features           | dataframe, default, trash, which, zip     |
| installed_plugins  |                                           |

### Additional context

Possible Failing Script:
https://pastebin.com/7thJJf9n

**Comment by fdncred** [comment]
From MacOS, I can run `^open /Applications/Discord.app` and it launches without issue.

**Comment by ColdIce1605** [comment]
I've  tried that same issue.

**Comment by fdncred** [comment]
i googled this issue and there are tons of hits called "Discord Black Screen" which seems like what you're describing. I'm curious though why it's happening only with nushell.

**Comment by ColdIce1605** [comment]
Yes then I run bash to get to bash then run discord through bash it works fine.

**Comment by fdncred** [comment]
do you see any environment variables that are different from bash to nushell? trying to think what could cause this. it could also be caused because the linux discord launch script runs through bash with a bunch of stuff, looking for updates and such, maybe something is getting horked shelling out to bash to run that script?

**Comment by cararemixed** [comment]
I know a number of apps stop working for me if nu is set as my default shell. When changing it you need to log out and log back in to see the full effect it has. On macOS here, VSCode complains that it can't load the environemt (whatever that means) and for Sublime Text, Package Control stops working entirely. Same if I launch from `nu` but have a different shell set when starting the session.

I'll try diffing the environment variables sometimes but I'm pretty sure this is an issue with nu or at least nu's integration with macOS.

**Comment by jntrnr** [comment]
@cararemixed - I'm wondering if some of the issues are coming from apps assuming they're working with a bash or bash-like shell.

Yehuda had an issue with VSCode for wsl and fixed it using "the workaround is to not use chsh but instead hardcode it in the Terminal settings".

Hopefully we can do a combination of fixing issues on our side as well as fixing issues in apps that assume functionality in the shell that it might not have.

**Comment by jntrnr** [comment]
@ColdIce1605 - for the original issue you reported, that script should get run through bash if it ends in a `.sh`. Nushell doesn't really sniff the hashbang, but instead uses the file extension to figure out how to run the file. It could be bash isn't where it expects it to be, or that the file extension is something it doesn't expect.

**Comment by ColdIce1605** [comment]
ok, i've found what is executed when launching discord its
command:
/usr/bin/discord

**Comment by fdncred** [comment]
What happens if you just run that /usr/bin/discord?

**Comment by ColdIce1605** [comment]
same as before

**Comment by ColdIce1605** [comment]
Also I think I found another app that doesn't like nushell, unityhub

**Comment by ColdIce1605** [comment]
@fdncred  @jntrnr ping?

**Comment by fdncred** [comment]
what does discord say about it? I'm wondering if certain env vars aren't being set.

**Comment by ColdIce1605** [comment]
https://pastebin.com/fZy0Lb5W
nothing useful

**Comment by ColdIce1605** [comment]
found the issue user error lol

**Comment by ColdIce1605** [comment]
actually it doesn't work when launching from the taskbar

**Comment by ColdIce1605** [comment]
yeah nevermind that libva errror is important as when launching from taskbar I have that same error again

**Comment by ColdIce1605** [comment]
@fdncred

**Comment by fdncred** [comment]
i'm not sure how to fix discord errors like this

**Comment by ColdIce1605** [comment]
Its because when running through the taskbar it's not getting the environment vars for some reason running through the command line works just fine.
